src/tests/ie_test_utils/common_test_utils/data_utils.cpp

```diff
@@ -202,7 +205,132 @@ size_t byte_size(const InferenceEngine::TensorDesc &tdesc) {
+    auto tensor_dims = tensor.get_shape();
+    auto n_dims = tensor_dims.size();
+    IE_ASSERT(values_dims.size() <= n_dims);
+    IE_ASSERT(n_dims <= MAX_N_DIMS);
+
+    SizeVector src_dims(MAX_N_DIMS, 1);
+    std::copy(values_dims.rbegin(), values_dims.rend(), src_dims.rbegin());
+
+    SizeVector dst_dims(MAX_N_DIMS, 1);
+    std::copy(tensor_dims.rbegin(), tensor_dims.rend(), dst_dims.rbegin());
+
+    bool compatible = true;
+    for (int i = 0; i < MAX_N_DIMS; i++) {
+        if (src_dims[i] != dst_dims[i] && src_dims[i] != 1)
+            compatible = false;
+    }
+
+    OPENVINO_ASSERT(compatible);
+
+    auto fill_strides_like_plain = [] (SizeVector dims) {
+        SizeVector str(dims.size());
+        if (str.empty())
+            return str;
+        else
+            str.back() = 1;
+
+        // stride[i] = stride[i+1]*d[i+1]
+        std::transform(dims.rbegin(), dims.rend() - 1, str.rbegin(), str.rbegin() + 1,
+                       [] (size_t d, size_t s) { return d * s; });
+
+        // zeroing broadcast dimension equal 1
+        std::transform(str.begin(), str.end(), dims.begin(), str.begin(),
+                       [] (size_t s, size_t d) { return d == 1 ? 0 : s; });
+
+        return str;
+    };
+
+    SizeVector src_strides = fill_strides_like_plain(src_dims);
+    SizeVector dst_strides = fill_strides_like_plain(dst_dims);
 
+    auto dst_ptr = tensor.data();
+    auto src_ptr = values.data();
+
+    using ov::element::Type_t;
+    switch (tensor.get_element_type()) {
+        case Type_t::u64:
+        case Type_t::i64:
+            copy_7D<uint64_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
+            break;
+        case Type_t::f32:
+        case Type_t::i32:
+            copy_7D<uint32_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
+            break;
+        case Type_t::i16:
+        case Type_t::u16:
+        case Type_t::f16:
+        case Type_t::bf16:
+            copy_7D<uint16_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
+            break;
+        case Type_t::u8:
+        case Type_t::i8:
+            copy_7D<uint8_t>(src_ptr, src_strides, dst_ptr, dst_strides, dst_dims);
+            break;
+        default:
+            OPENVINO_THROW("Unsupported precision by fill_data_with_broadcast function");
+    }
+}
+
+template<ov::element::Type_t SRC_E, ov::element::Type_t DST_E>
+void copy_with_convert(ov::Tensor& src_tensor, ov::Tensor& dst_tensor) {
+    using SRC_TYPE = typename ov::fundamental_type_for<SRC_E>;
+    using DST_TYPE = typename ov::fundamental_type_for<DST_E>;
+
+    auto src_ptr = src_tensor.data<SRC_TYPE>();
+    auto src_size = src_tensor.get_size();
+
+    auto dst_ptr = dst_tensor.data<DST_TYPE>();
+
+    std::copy(src_ptr, src_ptr + src_size, dst_ptr);
+}
+
+ov::Tensor make_with_precision_convert(ov::Tensor& tensor, ov::element::Type_t prc) {
+    OPENVINO_ASSERT(tensor.get_element_type() == ov::element::Type_t::f32);
```

Comment:
Why do you support only `f32` element type?

Reply:
This func may be called only from 1 place. And in this case tensor type is f32